### PR TITLE
feat(FIN-1204): Add transaction-auditor-diffs topic to billing namespace

### PR DESCRIPTION
**Pull request by lucassouzavieira**
It also includes configuration for producer and consumers.

### CODE REVIEW

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-billing
Path: dev-aws/kafka-shared-msk/billing
Commit ID: de4b1e73794326e2a5c45fc48d3ba61577d7e5db
Requested At: 2024-10-03T14:21:00Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-billing
Path: prod-aws/kafka-shared-msk/billing
Commit ID: 7d88a7e109f84a872e346f5f8bef3b3b3559910a
Requested At: 2024-10-03T14:21:01Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/billing` as comment.

File: dev-aws/kafka-shared-msk/billing/transaction-auditor-events.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "compression.type" = "zstd"
+    # 805 GB
+    "retention.bytes" = "805306368000"
+    # 105 MB
+    "max.message.bytes" = "104857600"
+    # Use tiered storage
+    "remote.storage.enable" = "true"
+    # keep data in hot storage for 2 days
+    "local.retention.ms" = "172800000"
+    "retention.ms"       = "5184000000"
```
Comment (sbuliarca) [inline dev-aws/kafka-shared-msk/billing/transaction-auditor-events.tf:15]: Please also put a comment to this value with the human readable value

File: prod-aws/kafka-shared-msk/billing/transaction-auditor-events.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "compression.type" = "zstd"
+    # 805 GB
+    "retention.bytes" = "805306368000"
+    # 105 MB
+    "max.message.bytes" = "104857600"
+    # Use tiered storage
+    "remote.storage.enable" = "true"
+    # keep data in hot storage for 2 days
+    "local.retention.ms" = "172800000"
+    "retention.ms"       = "5184000000"
```
Comment (sbuliarca) [inline prod-aws/kafka-shared-msk/billing/transaction-auditor-events.tf:15]: Same here.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: pubsub/kafka-shared-msk-billing
Path: dev-aws/kafka-shared-msk/billing
Commit ID: e93699a1abb1c7383703ff0dd9ed4ea6771abbe6
Requested At: 2024-10-03T14:39:41Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/kafka-shared-msk/billing` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: pubsub/kafka-shared-msk-billing
Path: prod-aws/kafka-shared-msk/billing
Commit ID: e93699a1abb1c7383703ff0dd9ed4ea6771abbe6
Requested At: 2024-10-03T14:39:41Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/kafka-shared-msk/billing` as comment.

**Review by sbuliarca — APPROVED** [review]

File: prod-aws/kafka-shared-msk/billing/transaction-auditor-events.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    # 805 GB
+    "retention.bytes" = "805306368000"
+    # 105 MB
+    "max.message.bytes" = "104857600"
+    # Use tiered storage
+    "remote.storage.enable" = "true"
+    # keep data in hot storage for 2 days
+    "local.retention.ms" = "172800000"
+    # keep data for 60 days
+    "retention.ms"       = "5184000000"
```
Comment (sbuliarca) [inline prod-aws/kafka-shared-msk/billing/transaction-auditor-events.tf:16]: This needs terraform formatting. 
For handling this automatically see: https://github.com/utilitywarehouse/kafka-cluster-config/blob/main/CONTRIBUTING.md

**Review by sbuliarca — APPROVED** [review]

**Pull request merged**